src/components/Sep8Send/Sep8Review.tsx

```diff
@@ -0,0 +1,132 @@
+  const { account, sep8Send, settings } = useRedux(
+    "account",
+    "sep8Send",
+    "settings",
+  );
+  const [submittedTx, setSubmittedTx] = useState<Transaction | undefined>();
+  const [revisedTx, setRevisedTx] = useState<Transaction | undefined>();
+  const [isApproved, setIsApproved] = useState(false);
+  const dispatch = useDispatch();
+
+  // user interaction handlers
+  const handleSubmitPayment = () => {
+    if (sep8Send.data.revisedTransaction.revisedTxXdr && isApproved) {
+      dispatch(sep8SubmitRevisedTransactionAction());
+    }
+  };
+
+  // use effect: complete action, close modal and refresh account balances
+  useEffect(() => {
+    if (sep8Send.status === ActionStatus.SUCCESS && account.data?.id) {
+      dispatch(
+        fetchAccountAction({
+          publicKey: account.data.id,
+          secretKey: account.secretKey,
+        }),
+      );
+      onClose();
+    }
+  }, [account.data?.id, account.secretKey, sep8Send.status, dispatch, onClose]);
+
+  // use effect: parse transaction XDRs
+  const { revisedTxXdr, submittedTxXdr } = sep8Send.data.revisedTransaction;
+  useEffect(() => {
+    const networkPassphrase = getNetworkConfig(settings.pubnet).network;
+
+    if (submittedTxXdr) {
+      const tx = TransactionBuilder.fromXDR(
+        submittedTxXdr,
+        networkPassphrase,
+      ) as Transaction;
+      setSubmittedTx(tx);
+    }
+
+    if (revisedTxXdr) {
+      const tx = TransactionBuilder.fromXDR(
+        revisedTxXdr,
+        networkPassphrase,
+      ) as Transaction;
+      setRevisedTx(tx);
+    }
+  }, [revisedTxXdr, submittedTxXdr, settings.pubnet]);
+
+  const renderSendPayment = () => (
+    <>
+      <Heading2 className="ModalHeading">
+        Review & Submit SEP-8 Transaction
+      </Heading2>
+
+      <div className="ModalMessage">
+        <p>
+          Please review the updated operations before submitting your SEP-8
+          payment.
+        </p>
+      </div>
+
+      {submittedTx?.operations && (
+        <>
+          <label className="ModalLabel">Original transaction operations</label>
+          <Json src={submittedTx.operations} />
+        </>
+      )}
+
+      {revisedTx?.operations && (
+        <>
+          <label className="ModalLabel">Revised transaction operations</label>
+          <Json src={revisedTx.operations} />
+        </>
+      )}
+
+      {sep8Send.errorString && (
+        <div className="ModalMessage error">
+          <p>{sep8Send.errorString}</p>
+        </div>
+      )}
+
+      <div className="ConfigurationItem ModalLabel">
+        <label htmlFor="claimable-balance-supported">
+          I approve executing these operations.
+        </label>
+        <Toggle
+          id="claimable-balance-supported"
+          checked={isApproved}
+          onChange={() => {
+            setIsApproved(!isApproved);
+          }}
+          disabled={sep8Send.status === ActionStatus.PENDING}
+        />
```

Comment:
We could show a checkbox instead of the toggle. We have it on the Account Viewer when generating a new keypair.

![image](https://user-images.githubusercontent.com/7346473/118553944-f616cc80-b72e-11eb-9556-b68a4e6d4fc2.png)

Reply:
Thanks for the suggestion. I've just tried using it but found an odd behavior, when I mark the checkbox the modal content gets out of bounds:

![Screen Shot 2021-05-17 at 18 23 21](https://user-images.githubusercontent.com/1952597/118558493-16e61e80-b73d-11eb-8b7a-4346c778e68e.png)


https://user-images.githubusercontent.com/1952597/118558502-1a79a580-b73d-11eb-81e1-f23564cacfad.mov

Reply:
Done it in https://github.com/stellar/stellar-demo-wallet/pull/157/commits/bb3adb25ee926d044370cdc59335c64c90fcf60b.